sympy/polys/euclidtools.py

```diff
@@ -1248,6 +1248,579 @@ def dup_zz_heu_gcd(f, g, K):
     raise HeuristicGCDFailed('no luck')
 
 
+def sparse_coeffs(p1, syms):
+    """
+    It converts a polynomial p1 into a sparse representation where the
+    monomials are split into parts with symbols and parts without symbols,
+    and the coefficients are stored accordingly.
```

Comment:
This explanation is not very clear. Also I think that the function name is confusing because what is returned is not what would usually be referred to as coefficients.

What this does is to take a polynomial from a multivariate ring like `K[x1, x2, y1, y2]` where syms is `{y1, y2}` and then treat it as a polynomial in `K[x1, x2][y1, y2]`. The coefficients are the elements of `K[x1, x2]` for the monomials made by the generators `y1` and `y2`.

In other contexts a similar operation is referred to as "dropping" generators e.g.:
```python
In [10]: R = QQ[x, y, z, t]

In [11]: R
Out[11]: QQ[x,y,z,t]

In [12]: R.drop(z, t)
Out[12]: QQ[x,y]

In [28]: p = R.from_sympy(x**2 + y**2 + z**2 + t**2)

In [29]: p
Out[29]: x**2 + y**2 + z**2 + t**2

In [30]: p.ring
Out[30]: Polynomial ring in x, y, z, t over QQ with lex order

In [31]: p.drop_to_ground(3)
Out[31]: x**2 + y**2 + z**2 + t**2

In [32]: p.drop_to_ground(3).ring
Out[32]: Polynomial ring in x, y, z over QQ[t] with lex order
```
Maybe it would be better to extend those methods of `PolyElement` so that they can do the operation here.